storage/gossip_test.go

```diff
@@ -42,15 +42,17 @@ func TestGossipFirstRange(t *testing.T) {
 
 	errors := make(chan error)
 	descs := make(chan *roachpb.RangeDescriptor)
-	tc.Servers[0].Gossip().RegisterCallback(gossip.KeyFirstRangeDescriptor,
+	unregister := tc.Servers[0].Gossip().RegisterCallback(gossip.KeyFirstRangeDescriptor,
 		func(_ string, content roachpb.Value) {
 			var desc roachpb.RangeDescriptor
 			if err := content.GetProto(&desc); err != nil {
 				errors <- err
 			} else {
 				descs <- &desc
 			}
-		})
+		},
+	)
+	defer unregister()
```

Comment:
Isn't this still flaky, at least in theory? You read `errors` for the last time in  line 105, so anytime after that a firing callback will still lose a goroutine.